dpsim/src/KLUAdapter.cpp

```diff
@@ -229,4 +210,53 @@ void KLUAdapter::printMatrixMarket(SparseMatrix &matrix, int counter) const
     }
     ofs.close();
 }
+
+void KLUAdapter::applyConfiguration()
+{
+	switch(mConfiguration.getScalingMethod())
+	{
+		case SCALING_METHOD::NO_SCALING:
+			mCommon.scale = 0;
+			break;
+		case SCALING_METHOD::SUM_SCALING:
+			mCommon.scale = 1;
+			break;
+		case SCALING_METHOD::MAX_SCALING:
+			mCommon.scale = 2;
+			break;
+		default:
+			mCommon.scale = 1;
+	}
+
+	// TODO: implement support for COLAMD (modifiy SuiteSparse)
+	switch(mConfiguration.getFillInReductionMethod())
+	{
+		case FILL_IN_REDUCTION_METHOD::AMD:
+			mPreordering = AMD_ORDERING;
+			break;
+		case FILL_IN_REDUCTION_METHOD::AMD_NV:
+			mPreordering = AMD_ORDERING_NV;
+			break;
+		case FILL_IN_REDUCTION_METHOD::AMD_RA:
+			mPreordering = AMD_ORDERING_RA;
+			break;
+		default:
+			mPreordering = AMD_ORDERING;
+	}
+
+	// NOTE: in case more partial refactorization methods are defined/developed, that are not implemented in KLU, this assigment would be invalid
+	mPartialRefactorizationMethod = mConfiguration.getPartialRefactorizationMethod();
+
+	switch(mConfiguration.getBTF())
+	{
+		case USE_BTF::DO_BTF:
+			mCommon.btf = 1;
+			break;
+		case USE_BTF::NO_BTF:
+			mCommon.btf = 0;
+			break;
+		default:
+			mCommon.btf = 1;
+	}
```

Comment:
Add logging of the applied configurations (use info level). Implement String conversions in `DirectLinearSolverConfiguration` and reuse here (e.g. `mConfiguration.getBTFString()` used here with SPD_LOGGER_INFO)?

Reply:
done